packages/tasit-action/src/TransactionSubscription.js

```diff
@@ -11,19 +11,46 @@ export class TransactionSubscription extends Subscription {
   #txConfirmations;
   #timeout;
   #lastConfirmationTime;
-
-  constructor(txPromise, provider) {
+  #contract;
+
+  // Contract passing itself to here, enables
+  // TransactionSubscription calling this.#contract._emitErrorEvent
+  // to emit error to the contract catch-all error listener
```

Comment:
I need to think about this more, but would it be cleaner for the `TransactionSubscription` to emit an error event on the `TransactionSubscription` event emitter that is listened to by the Contract which then emits its own error event relaying that original error payload? So in other words, a contract would always be listening to error events on any event emitters of things that belong to it

Reply:
Oh, it seems an excellent way to go. I'll do some tryouts.

Reply:
Cool! The only part I'd need to think about more is how it works if there are multiple TransactionSubscriptions active

Reply:
Yep, I think so. So the contract would just need to be automatically listening for errors on all active `TransactionsSubscription`'s to pass them through to the `Contract` error event.

Reply:
When the contract is listening for an error event, and then the user calls 
 `subscribedEventNames()` he gets ["error"] even he didn't registered one:

```javascript
      txSubscription = simpleStorage.setValue("hello world");

      // AssertionError: expected [ 'error' ] to be empty 
      expect(txSubscription.subscribedEventNames()).to.be.empty; 

      txSubscription.on("confirmation", listener);
      txSubscription.off("confirmation");

      // AssertionError: expected [ 'error' ] to be empty
      expect(txSubscription.subscribedEventNames()).to.be.empty; 
```

Reply:
Still open, I've kept the first approach when I'm waiting your opinion.

Reply:
I suppose we could have that getter function not return error in the cases we don’t want it to. Or we could decide that in a sense error is a subscribed event name here as a result of listening at the contract level. Either is okay with me for this PR. Slight preference for the second option (and in that case just the test expectations would need to change)

Reply:
Yes, I think so. Maybe we should think more about this error behavior. I'll open a `question` issue with that and another question that I've thought about. Alright?

Reply:
Sounds good!

Reply:
I've created separated issues https://github.com/tasitlabs/TasitSDK/issues/163 and https://github.com/tasitlabs/TasitSDK/issues/164

Reply:
Thanks. With that, should we resolve this?

Reply:
I think so, resolving.